### DISCUSSION TITLE
AreaChart and its color problem

### DISCUSSION THREAD

**Discussion by artemklishch** [recharts/recharts#3462] [Q&A] [answered]
Hello!
When I render the "AreaChart" chart in list via "map" method I meet the problem.
The array contains the different colors for every item. But when charts are rendered - they have the color of the first item of array (but every item certainly has its own different color). 
Only lines (I mean - strokes) themselves have correct colors, but area below lines - the colors of the first item.
I can't explain it and can't resolve the issue. I suspect that the tag "defs" by some way remembers the color value that the first item has, and then it is not updated on next iterations.
Because renderings the list of another chart types of the "Recharts" library don't have this problem. And lines themselves as I said, have correct colors.
Has anybody met the same case? 
Could anybody advise how to resolve it?
Thank you!

**Comment by ckifer** [comment]
Hi, 

Can you represent your issue in this codesandbox template?
https://codesandbox.io/s/simple-area-chart-forked-s1q6si

**Reply by ckifer** [reply]
Seems to be working here
https://codesandbox.io/s/simple-area-chart-forked-78pmoj?file=/src/App.tsx

**Reply by ckifer** [reply]
Seems to work for me

https://codesandbox.io/s/simple-area-chart-forked-v91ilx

**Reply by ckifer** [reply]
Can you modify the sandbox to represent your exact problem? I don't see any gradients in the example

**Reply by artemklishch** [reply]
I am sorry
I couldn't insert the correct link here

This one:
https://codesandbox.io/s/simple-area-chart-forked-jcgrnx?file=/src/App.tsx

with linear gradient

**Reply by ckifer** [reply]
Here you are https://codesandbox.io/s/simple-area-chart-forked-95v887

Same issue as #3461 

You need a unique ID for each linear gradient

**Reply by artemklishch** [reply]
Thank you so much!

**Reply by ckifer** [reply]
np!

**Comment by ckifer** [comment, marked as answer]
Answering here to mark as answer

https://codesandbox.io/s/simple-area-chart-forked-95v887

Same issue as https://github.com/recharts/recharts/issues/3461

OP needed a unique ID for each linear gradient created in a loop
